Frameworks/Foundation/NSDecimal.mm

```diff
@@ -0,0 +1,647 @@
+//******************************************************************************
+//
+// Copyright (c) Microsoft. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#import <Starboard.h>
+#import <StubReturn.h>
+#import <Foundation/NSDecimal.h>
+#import <algorithm>
+
+#define IS_NEGATIVE(decimal) (decimal)->_isNegative
+
+#define IS_POSITIVE(decimal) !IS_NEGATIVE(decimal)
+
+#define IS_ZERO(decimal) (((decimal)->_length == 0) && IS_POSITIVE(decimal))
+
+#define INVALIDATE_DECIMAL(decimal) \
+    \
+decimal->_length = 0;               \
+    \
+decimal->_isNegative = YES;
+
+#define SET_TO_ZERO(decimal) \
+    \
+decimal->_length = 0;        \
+    \
+decimal->_isNegative = NO;   \
+    \
+NSDecimalCompact(decimal);
+
+#define VALIDATE_ARITHMETIC_INPUT(result, leftOperand, rightOperand)                 \
+    if (NSDecimalIsNotANumber(leftOperand) || NSDecimalIsNotANumber(rightOperand)) { \
+        INVALIDATE_DECIMAL(result);                                                  \
+        return NSCalculationOverflow;                                                \
+    }
+
+const int _MAX_EXP = 127;
```

Comment:
why won't fix? This is against CC and is all new code. Lets make the effort to be compliant.

---
In reply to: [81209420](https://github.com/Microsoft/WinObjC/pull/1032#discussion_r81209420) [](ancestors = 81209420)